modules/JsonExporter/src/main/java/uk/ac/ox/oii/jsonexporter/JSONExporter.java

```diff
@@ -0,0 +1,277 @@
+import org.gephi.utils.longtask.spi.LongTask;
+import org.gephi.utils.progress.Progress;
+import org.gephi.utils.progress.ProgressTicket;
+import uk.ac.ox.oii.jsonexporter.model.GraphEdge;
+import uk.ac.ox.oii.jsonexporter.model.GraphElement;
+import uk.ac.ox.oii.jsonexporter.model.GraphNode;
+
+public class JSONExporter implements GraphExporter, LongTask, CharacterExporter {
+
+    private boolean exportVisible = false;
+    private Workspace workspace;
+    private Writer writer;
+    private ProgressTicket progress;
+    private boolean cancel = false;
+    //private File path;
+
+    @Override
+    public boolean execute() {
+        try {
+            if (writer != null) {//path.getParentFile().exists()
+                GraphModel graphModel = workspace.getLookup().lookup(GraphModel.class);
+                Graph graph = null;
+                if (exportVisible) {
+                    graph = graphModel.getGraphVisible();
+                } else {
+                    graph = graphModel.getGraph();
+                }
+
+                //Count the number of tasks (nodes + edges) and start the progress
+                int tasks = graph.getNodeCount() + graph.getEdgeCount();
+                Progress.start(progress, tasks);
+
+                //FileWriter fwriter = new  FileWriter(writer);
+
+                Gson gson = new Gson();
+                EdgeColor colorMixer = new EdgeColor(EdgeColor.Mode.MIXED);
+
+                //HashMap<String, String> nodeIdMap = new HashMap<String, String>();
+                //int nodeId = 0;
+                //EdgeColor colorMixer = new EdgeColor(EdgeColor.Mode.MIXED);
+                //Write data.json
+
+
+                Table attModel = graphModel.getNodeTable();
+                HashSet<GraphElement> jNodes = new HashSet<GraphElement>();
+                Node[] nodeArray = graph.getNodes().toArray();
+                
+                for (Node n : nodeArray) {
+
+                    String id = n.getId().toString();
+                    String label = n.getLabel();
+                    float x = n.x();
+                    float y = n.y();
+                    System.out.println(n.z());
+                    float size = n.size();
+                    String color = "rgb(" + (int) (n.r() * 255) + "," + (int) (n.g() * 255) + "," + (int) (n.b() * 255) + ")";
+
+                    /*if (renumber) {
+                     String newId=String.valueOf(nodeId);
+                     nodeIdMap.put(id,newId); 
+                     id=newId;
+                     nodeId++;
+                     }*/
+
+                    GraphNode jNode = new GraphNode(id);
+                    jNode.setLabel(label);
+                    jNode.setX(x);
+                    jNode.setY(y);
+                    jNode.setSize(size);
+                    jNode.setColor(color);
+
+                    
+                    for (Column col : attModel) {
+                            String name = col.getTitle();
```

Comment:
Unlike the column's id, the column's title might be localized so it isn't recommended to do comparaison based on that. I would suggest to rather use `col.getId()`. The default ID column's id is `id` and same for `label`.